.changeset/yellow-suits-smell.md

```diff
@@ -0,0 +1,5 @@
+---
+"@primer/components": patch
+---
+
+Migrate `Header` component to TypeScript
```

Comment:
[minor] We haven't been including the word `component` in our release notes for these migrations. For consistency let's remove it here:
```suggestion
Migrate `Header` to TypeScript
```